base/array.jl

```diff
@@ -711,8 +711,8 @@ function find(testf::Function, A::AbstractArray)
     # use a dynamic-length array to store the indexes, then copy to a non-padded
     # array for the return
     tmpI = Array(Int, 0)
-    for i = 1:length(A)
-        if testf(A[i])
+    for (i,j) = enumerate(eachindex(A))
+        if testf(A[j])
```

Comment:
Because some package might implement a non-standard Fortran-style indexing, where indices have an offset and don't start at 1.